sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -353,7 +353,7 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
   test("foreachPartition") {
     val ds = Seq(("a", 1), ("b", 2), ("c", 3)).toDS()
     val acc = sparkContext.longAccumulator
-    ds.foreachPartition(_.foreach(v => acc.add(v._2)))
+    ds.foreachPartition((it: Iterator[(String, Int)]) => it.foreach(v => acc.add(v._2)))
```

Comment:
Actually, no not the same issue as https://issues.apache.org/jira/browse/SPARK-14643 but similar. Let me ask the question there about how much source compatibility matters across Scala versions, _for Java_.

I'm not in any event proposing to enable a 2.12 build anytime soon, just see if we can get it closer without any implications for 2.11